--- api/web/src/components/CloudTAK/MainMenuContents.vue	2025-12-15 00:55:18.036815047 +0000
+++ api/web/src/components/CloudTAK/MainMenuContents.vue	2025-12-15 00:55:18.032068626 +0000
@@ -184,42 +184,7 @@
         <div
             v-else-if='["home", "home-menu"].includes(String(route.name))'
         >
-            <div class='d-flex justify-content-center mb-2'>
-                <TablerDropdown
-                    position='right'
-                >
-                    <template #default>
-                        <TablerIconButton
-                            title='Application Switcher'
-                            class='hover-button'
-                            :hover='false'
-                        >
-                            <IconGridDots
-                                :size='32'
-                                stroke='1'
-                            />
-                        </TablerIconButton>
-                    </template>
-                    <template #dropdown>
-                        <div class='card'>
-                            <div class='card-body'>
-                                <div
-                                    class='px-2 py-2 d-flex align-items-center hover rounded cursor-pointer'
-                                    @click='external("/video")'
-                                >
-                                    <IconDeviceTv
-                                        size='32'
-                                        stroke='1'
-                                    />
-                                    <div class='mx-2'>
-                                        Video Wall
-                                    </div>
-                                </div>
-                            </div>
-                        </div>
-                    </template>
-                </TablerDropdown>
-            </div>
+
             <div class='d-flex justify-content-center mb-2'>
                 <div class='position-relative'>
                     <img
@@ -276,10 +241,8 @@
     IconMessage,
     IconNetwork,
     IconPackages,
-    IconGridDots,
     IconSettings,
-    IconDeviceTv,
-    IconAmbulance,
+    IconReplace,
     IconServerCog,
     IconBoxMultiple,
     IconFileImport,
@@ -288,7 +251,6 @@
     IconLayoutList,
 } from '@tabler/icons-vue';
 import {
-    TablerDropdown,
     TablerIconButton,
     TablerInput,
     TablerNone,
@@ -380,7 +342,7 @@
         route: '/menu/missions',
         tooltip: 'Data Sync',
         description: 'Real-Time Datasets',
-        icon: IconAmbulance,
+        icon: IconReplace,
     },
     {
         key: 'packages',
