diff --git a/api/web/src/stores/map.ts b/api/web/src/stores/map.ts
index 41009cf42..48f651caa 100644
--- a/api/web/src/stores/map.ts
+++ b/api/web/src/stores/map.ts
@@ -14,6 +14,8 @@ import IconManager from './modules/icons.ts';
 import * as Comlink from 'comlink';
 import AtlasWorker from '../workers/atlas.ts?worker&url';
 import COT from '../base/cot.ts';
+import type { DatabaseType } from '../base/database.ts';
+import { db } from '../base/database.ts';
 import { WorkerMessageType, LocationState } from '../base/events.ts';
 import type { WorkerMessage } from '../base/events.ts';
 import Overlay from '../base/overlay.ts';
@@ -33,6 +35,8 @@ export const useMapStore = defineStore('cloudtak', {
         _map?: mapgl.Map;
         _draw?: DrawTool;
         _icons?: IconManager;
+
+        db: DatabaseType;
         channel: BroadcastChannel;
 
         toImport: Feature[]
@@ -48,6 +52,11 @@ export const useMapStore = defineStore('cloudtak', {
         manualLocationMode: boolean;
         gpsWatchId: number | null;
 
+        toastOffset: {
+            x: number;
+            y: number;
+        };
+
         permissions: {
             location: boolean;
             notification: boolean;
@@ -56,7 +65,6 @@ export const useMapStore = defineStore('cloudtak', {
         worker: Comlink.Remote<Atlas>;
         mission: Subscription | undefined;
         mapConfig: MapConfig;
-        notifications: Array<TAKNotification>;
         container?: HTMLElement;
         hasTerrain: boolean;
         hasNoChannels: boolean;
@@ -97,13 +105,14 @@ export const useMapStore = defineStore('cloudtak', {
             callsign: 'Unknown',
             toImport: [],
             location: LocationState.Loading,
+            db,
             channel: new BroadcastChannel("cloudtak"),
             zoom: 'conditional',
             distanceUnit: 'meter',
+            toastOffset: { x: 70, y: 10 },
             manualLocationMode: false,
             gpsWatchId: null,
             locked: [],
-            notifications: [],
             hasTerrain: false,
             hasNoChannels: false,
             isTerrainEnabled: false,
@@ -191,7 +200,13 @@ export const useMapStore = defineStore('cloudtak', {
 
             await overlay.delete();
             if (overlay.mode === 'mission' && overlay.mode_id) {
-                await this.worker.db.subscriptionDelete(overlay.mode_id);
+                const sub = await Subscription.from(overlay.mode_id, localStorage.token, {
+                    subscribed: true
+                });
+
+                if (sub) {
+                    await sub.update({ subscribed: false });
+                }
             }
         },
         makeActiveMission: async function(mission?: Subscription): Promise<void> {
@@ -273,19 +288,11 @@ export const useMapStore = defineStore('cloudtak', {
         },
 
         /**
-         * Trigger a rerender of the underlyin GeoJSON Features
+         * Trigger a rerender of the underlying GeoJSON Features
          */
         refresh: async function(): Promise<void> {
             await this.updateCOT();
-
-            const missions = [];
-
-            for (const uid of await this.worker.db.subscriptionListUid({ dirty: true })) {
-                missions.push(this.loadMission(uid));
-            }
-            await Promise.allSettled(missions);
         },
-
         updateCOT: async function(): Promise<void> {
             try {
                 const diff = await this.worker.db.diff();
@@ -334,16 +341,16 @@ export const useMapStore = defineStore('cloudtak', {
             const oStore = this.map.getSource(String(overlay.id));
             if (!oStore) return false
 
-            let sub = await this.worker.db.subscriptionGet(guid);
-
-            if (!sub) {
-                sub = await this.worker.db.subscriptionLoad(guid, overlay.token || undefined)
-            }
+            const sub = await Subscription.load(guid, {
+                token: localStorage.token,
+                subscribed: true,
+                missiontoken: overlay.token || undefined
+            });
 
             // @ts-expect-error Source.setData is not defined
-            oStore.setData(await sub.collection());
+            oStore.setData(await sub.feature.collection());
 
-            await this.worker.db.subscriptionClean(guid);
+            await sub.update({ dirty: false });
 
             return true;
         },
@@ -356,12 +363,20 @@ export const useMapStore = defineStore('cloudtak', {
                 const msg = event.data;
 
                 if (!msg || !msg.type) return;
-                if (msg.type === WorkerMessageType.Map_FlyTo) {
+                if (msg.type === WorkerMessageType.Map_FitBounds) {
                     if (msg.body.options.speed === null) {
                         msg.body.options.speed = Infinity;
                     }
 
                     map.fitBounds(msg.body.bounds, msg.body.options);
+                } else if (msg.type === WorkerMessageType.Map_FlyTo) {
+                    if (msg.body.speed === null) {
+                        msg.body.speed = Infinity;
+                    }
+
+                    if (!msg.body.zoom) msg.body.zoom = this.map.getZoom();
+
+                    map.flyTo(msg.body);
                 } else if (msg.type === WorkerMessageType.Profile_Location_Source) {
                     this.location = msg.body.source as LocationState;
                 } else if (msg.type === WorkerMessageType.Profile_Callsign) {
@@ -382,11 +397,6 @@ export const useMapStore = defineStore('cloudtak', {
                     this.hasNoChannels = true;
                 } else if (msg.type === WorkerMessageType.Channels_List) {
                     this.hasNoChannels = false;
-                } else if (msg.type === WorkerMessageType.Notification) {
-                    this.notifications.push({
-                        ...msg.body,
-                        created: new Date().toISOString()
-                    } as TAKNotification);
                 } else if (msg.type === WorkerMessageType.Mission_Change_Feature) {
                     this.loadMission(msg.body.guid);
                 } else if (msg.type === WorkerMessageType.Contact_Change) {
@@ -684,7 +694,7 @@ export const useMapStore = defineStore('cloudtak', {
                 const basemaps = await std(burl) as APIList<Basemap>;
 
                 if (basemaps.items.length > 0) {
-                    const basemap = await Overlay.create(map, {
+                    const basemap = await Overlay.create({
                         name: basemaps.items[0].name,
                         pos: -1,
                         type: 'raster',
@@ -699,7 +709,6 @@ export const useMapStore = defineStore('cloudtak', {
 
             for (const item of profileOverlays.items) {
                 this.overlays.push(await Overlay.create(
-                    map,
                     item as ProfileOverlay,
                     {
                         skipSave: true
@@ -707,7 +716,7 @@ export const useMapStore = defineStore('cloudtak', {
                 ));
             }
 
-            this.overlays.push(await Overlay.internal(map, {
+            this.overlays.push(await Overlay.internal({
                 id: -1,
                 name: 'CoT Icons',
                 type: 'geojson',
@@ -723,6 +732,7 @@ export const useMapStore = defineStore('cloudtak', {
                     try {
                         await this.loadMission(overlay.mode_id);
                     } catch (err) {
+                        console.error('Failed to load Mission', err)
                         // TODO: Handle this gracefully
                         // The Mission Sync is either:
                         // - Deleted
