diff --git a/api/web/src/workers/atlas-database.ts b/api/web/src/workers/atlas-database.ts
index e8b2e6ba9..7d392ecbc 100644
--- a/api/web/src/workers/atlas-database.ts
+++ b/api/web/src/workers/atlas-database.ts
@@ -784,4 +784,16 @@ export default class AtlasDatabase {
 
         return new Set(list);
     }
+
+    /**
+     * Return a list of all CoTs in the store
+     */
+    async list(): Promise<Array<{ id: string, properties: { callsign?: string }, geometry: unknown, is_skittle: boolean }>> {
+        return Array.from(this.cots.values()).map(cot => ({
+            id: cot.id,
+            properties: { callsign: cot.properties.callsign },
+            geometry: cot.geometry,
+            is_skittle: cot.is_skittle
+        }));
+    }
 }
