diff --git a/api/web/src/components/CloudTAK/util/ChannelInfo.vue b/api/web/src/components/CloudTAK/util/ChannelInfo.vue
index 84dbbe088..a264d33ea 100644
--- a/api/web/src/components/CloudTAK/util/ChannelInfo.vue
+++ b/api/web/src/components/CloudTAK/util/ChannelInfo.vue
@@ -19,7 +19,7 @@ import {
 const props = defineProps({
     label: {
         type: String,
-        default: 'Missions'
+        default: 'Data Syncs'
     }
 });
 </script>
diff --git a/api/web/src/components/CloudTAK/util/EmptyInfo.vue b/api/web/src/components/CloudTAK/util/EmptyInfo.vue
index 8777f6b23..50af66f13 100644
--- a/api/web/src/components/CloudTAK/util/EmptyInfo.vue
+++ b/api/web/src/components/CloudTAK/util/EmptyInfo.vue
@@ -32,7 +32,7 @@
                 </TablerIconButton>
             </div>
         </template>
-        <template v-else-if='props.type === "Missions"'>
+        <template v-else-if='props.type === "Data Syncs"'>
             <IconAlertCircle
                 :size='32'
                 stroke='2'
diff --git a/api/web/src/components/CloudTAK/util/ShareToMission.vue b/api/web/src/components/CloudTAK/util/ShareToMission.vue
index 21eb016d9..923ad6a19 100644
--- a/api/web/src/components/CloudTAK/util/ShareToMission.vue
+++ b/api/web/src/components/CloudTAK/util/ShareToMission.vue
@@ -23,7 +23,7 @@
         <TablerLoading v-if='loading' />
         <EmptyInfo
             v-else-if='!missions.length'
-            type='Missions'
+            type='Data Syncs'
         />
         <template v-else>
             <div
