diff --git a/api/web/src/components/CloudTAK/util/ChannelInfo.vue b/api/web/src/components/CloudTAK/util/ChannelInfo.vue
index 84dbbe088..a264d33ea 100644
--- a/api/web/src/components/CloudTAK/util/ChannelInfo.vue
+++ b/api/web/src/components/CloudTAK/util/ChannelInfo.vue
@@ -19,7 +19,7 @@ import {
 const props = defineProps({
     label: {
         type: String,
-        default: 'Missions'
+        default: 'Data Syncs'
     }
 });
 </script>
