diff --git a/api/web/src/components/CloudTAK/util/ShareToMission.vue b/api/web/src/components/CloudTAK/util/ShareToMission.vue
index 46def8253..a83f173fd 100644
--- a/api/web/src/components/CloudTAK/util/ShareToMission.vue
+++ b/api/web/src/components/CloudTAK/util/ShareToMission.vue
@@ -28,7 +28,7 @@
             <TablerLoading v-if='loading' />
             <EmptyInfo
                 v-else-if='!missions.length'
-                type='Missions'
+                type='Data Syncs'
             />
             <template v-else>
                 <div
