--- a/api/lib/types.ts
+++ b/api/lib/types.ts
@@ -166,6 +166,7 @@ export const ServerResponse = Type.Object({
     api: Type.String(),
     webtak: Type.String(),
     auth: Type.Boolean({ "description": "Once an admin certificate is configured it is not retrivable. This boolean refers to if a certificate is currently loaded" }),
+    oidc_enabled: Type.Boolean({ "description": "Whether OIDC/SSO authentication is enabled" }),
     certificate: Type.Optional(Type.Object({
         subject: Type.String(),
         validFrom: Type.String(),
