diff --git a/api/nginx.conf.js b/api/nginx.conf.js
index a094f10e7..ead61a2de 100644
--- a/api/nginx.conf.js
+++ b/api/nginx.conf.js
@@ -23,7 +23,10 @@ if (url.hostname === 'localhost') {
         'worker-src': [`'self'`, 'blob:'],
         'style-src-elem': [`'self'`, `'unsafe-inline'`],
         'style-src-attr': [`'unsafe-inline'`],
-        'connect-src': [`'self'`]
+        'connect-src': [`'self'`],
+        // TEMPORARY: added to allow CloudTAK to be embedded in https://d2iy9yezumpf3t.cloudfront.net
+        // Remove the line below when embedding is no longer needed (revert to just 'connect-src' above)
+        'frame-ancestors': [`'self'`, 'https://d2iy9yezumpf3t.cloudfront.net']
     }
 
     cspstr = `add_header 'Content-Security-Policy' "`
@@ -79,6 +82,10 @@ http {
 
     access_log  /var/log/nginx/access.log  main;
 
+    # Increase buffer sizes for ALB OIDC cookies
+    large_client_header_buffers 4 32k;
+    client_header_buffer_size 32k;
+
     server {
         listen 5000;
         listen [::]:5000;
