diff --git a/api/derived-types.d.ts b/api/derived-types.d.ts
index 72d50e474..45e5a3229 100644
--- a/api/derived-types.d.ts
+++ b/api/derived-types.d.ts
@@ -344,6 +344,7 @@ export interface paths {
                         "application/json": {
                             id: number;
                             name: string;
+                            description?: unknown;
                         };
                     };
                 };
diff --git a/api/routes/agency.ts b/api/routes/agency.ts
index 1e8897dbf..7ce70b491 100644
--- a/api/routes/agency.ts
+++ b/api/routes/agency.ts
@@ -8,6 +8,7 @@ import * as Default from '../lib/limits.js';
 export const AgencyResponse = Type.Object({
     id: Type.Integer(),
     name: Type.String(),
+    description: Type.Optional(Type.Any()),
 });
 
 export default async function router(schema: Schema, config: Config) {
diff --git a/api/web/src/components/ETL/Connection/AgencyBadge.vue b/api/web/src/components/ETL/Connection/AgencyBadge.vue
index c034ebb89..5bba1cb7c 100644
--- a/api/web/src/components/ETL/Connection/AgencyBadge.vue
+++ b/api/web/src/components/ETL/Connection/AgencyBadge.vue
@@ -52,7 +52,7 @@
                             </div>
                             <div
                                 class='datagrid-content'
-                                v-text='"No Description"'
+                                v-text='agency.description || "No Description"'
                             />
                         </div>
                     </div>
